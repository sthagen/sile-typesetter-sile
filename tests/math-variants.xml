<sile>
<use module="packages.retrograde" target="v0.15.0" />
<use module="packages.math"/>

Math variants, text, MathML:
<mathml>
  <mrow>
  <mo>∀</mo><mi>n</mi><mo>∈</mo><mi mathvariant="double-struck">N</mi>
  <mo>,</mo> <mo>(</mo> <msup> <mi mathvariant="normal">e</mi>
    <mi>n</mi> </msup> <mo>,</mo> <mo mathvariant="double-struck">sin</mo>
  <mo>(</mo><mi>n</mi><mo>)</mo> <mo>)</mo>
  <mo>∈</mo>
  <msup>
    <mrow>
      <mo>(</mo><mi mathvariant="double-struck">R</mi>
      <mo>⧵</mo><mi mathvariant="double-struck">Q</mi><mo>)</mo>
    </mrow>
    <mn>2</mn>
  </msup>
  </mrow>
</mathml>

Math variants, text (large font), MathML:
<set parameter="math.font.size" value="18">
  <mathml>
    <mrow>
    <mo>∀</mo><mi>n</mi><mo>∈</mo><mi mathvariant="double-struck">N</mi>
    <mo>,</mo> <mo>(</mo> <msup> <mi mathvariant="normal">e</mi>
      <mi>n</mi> </msup> <mo>,</mo> <mo mathvariant="double-struck">sin</mo>
    <mo>(</mo><mi>n</mi><mo>)</mo> <mo>)</mo>
    <mo>∈</mo>
    <msup>
      <mrow>
        <mo>(</mo><mi mathvariant="double-struck">R</mi>
        <mo>⧵</mo><mi mathvariant="double-struck">Q</mi><mo>)</mo>
      </mrow>
      <mn>2</mn>
    </msup>
    </mrow>
  </mathml>
</set>

Math variants, display, MathML:
<mathml mode="display">
  <mrow>
  <mo>∀</mo><mi>n</mi><mo>∈</mo><mi mathvariant="double-struck">N</mi>
  <mo>,</mo> <mo>(</mo> <msup> <mi mathvariant="normal">e</mi>
    <mi>n</mi> </msup> <mo>,</mo> <mo mathvariant="double-struck">sin</mo>
  <mo>(</mo><mi>n</mi><mo>)</mo> <mo>)</mo>
  <mo>∈</mo>
  <msup>
    <mrow>
      <mo>(</mo><mi mathvariant="double-struck">R</mi>
      <mo>⧵</mo><mi mathvariant="double-struck">Q</mi><mo>)</mo>
    </mrow>
    <mn>2</mn>
  </msup>
  </mrow>
</mathml>

Math variants, display (large font), MathML:
<set parameter="math.font.size" value="18">
  <mathml mode="display">
    <mrow>
    <mo>∀</mo><mi>n</mi><mo>∈</mo><mi mathvariant="double-struck">N</mi>
    <mo>,</mo> <mo>(</mo> <msup> <mi mathvariant="normal">e</mi>
      <mi>n</mi> </msup> <mo>,</mo> <mo mathvariant="double-struck">sin</mo>
    <mo>(</mo><mi>n</mi><mo>)</mo> <mo>)</mo>
    <mo>∈</mo>
    <msup>
      <mrow>
        <mo>(</mo><mi mathvariant="double-struck">R</mi>
        <mo>⧵</mo><mi mathvariant="double-struck">Q</mi><mo>)</mo>
      </mrow>
      <mn>2</mn>
    </msup>
    </mrow>
  </mathml>
</set>

Math variants, text, TeX:
<math>
  \forall n \in \mi[mathvariant=double-struck]{N},
  (\mi[mathvariant=normal]{e}^n, \mo[mathvariant=normal]{sin}(n)) \in
  {(\mi[mathvariant=double-struck]{R} \setminus
  \mi[mathvariant=double-struck]{Q})}^2
</math>

Math variants, display, TeX:
<math mode="display">
  \forall n \in \mi[mathvariant=double-struck]{N},
  (\mi[mathvariant=normal]{e}^n, \mo[mathvariant=normal]{sin}(n)) \in
  {(\mi[mathvariant=double-struck]{R} \setminus
  \mi[mathvariant=double-struck]{Q})}^2
</math>

</sile>
