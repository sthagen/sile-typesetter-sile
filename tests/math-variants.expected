Set paper size 	595.275597	841.8897729
Begin page
Mx 	49.7638
My 	51.9745
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	48 68 87 75 w=21.6113	(Math)
Mx 	74.0466
T	89 68 85 76 68 81 87 86 w=33.6035	(variants)
Mx 	107.6501
T	15 w=2.2900	(,)
Mx 	112.6116
T	87 72 91 87 w=16.5332	(text)
Mx 	129.1448
T	15 w=2.2900	(,)
Mx 	134.1063
T	48 68 87 75 48 47 w=34.4336	(MathML)
Mx 	168.5399
T	29 w=2.2900	(:)
Mx 	173.5014
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	2070 w=5.9800	(nil)
Mx 	179.4814
T	2758 w=4.7100	(nil)
Mx 	187.0421
T	2078 w=6.3700	(nil)
Mx 	196.2627
T	1880 w=8.2400	(nil)
Mx 	204.5027
T	13 w=2.2000	(nil)
Mx 	208.3694
Mx 	3.0000
T	3798 a=4.2100	(nil)
Mx 	212.5794
T	70 w=4.4700	(nil)
Mx 	217.0494
My 	48.8745
Set font 	Libertinus Math;8;400;Regular;normal;;;LTR
T	2758 w=3.7328	(nil)
Mx 	221.2122
My 	51.9745
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	13 w=2.2000	(nil)
Mx 	225.0789
T	3000 2990 2995 w=14.3900	(nil)
Mx 	239.4689
Mx 	3.0000
T	3798 a=4.2100	(nil)
Mx 	243.6789
T	2758 w=4.7100	(nil)
Mx 	248.3889
Mx 	2.9900
T	3799 a=4.2100	(nil)
Mx 	252.5989
Mx 	2.9900
T	3799 a=4.2100	(nil)
Mx 	259.6595
T	2078 w=6.3700	(nil)
Mx 	268.8802
Mx 	2.8200
T	9 a=3.5600	(nil)
Mx 	272.4402
T	1887 w=6.1800	(nil)
Mx 	280.8716
T	2601 w=5.2700	(nil)
Mx 	288.3930
T	1884 w=7.0200	(nil)
Mx 	295.4130
Mx 	2.8200
T	10 a=3.5600	(nil)
Mx 	298.9730
My 	46.9745
Set font 	Libertinus Math;8;400;Regular;normal;;;LTR
T	19 w=3.7200	(nil)
Mx 	49.7638
My 	72.8485
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	48 68 87 75 w=21.6113	(Math)
Mx 	74.0295
T	89 68 85 76 68 81 87 86 w=33.6035	(variants)
Mx 	107.6330
T	15 w=2.2900	(,)
Mx 	112.5774
T	87 72 91 87 w=16.5332	(text)
Mx 	131.7649
T	11 w=3.1689	(()
Mx 	134.9339
T	79 68 85 74 72 w=20.7568	(large)
Mx 	158.3451
T	73 82 81 87 w=17.0996	(font)
Mx 	175.4447
T	12 w=3.1689	())
Mx 	178.6136
T	15 w=2.2900	(,)
Mx 	183.5580
T	48 68 87 75 48 47 w=34.4336	(MathML)
Mx 	217.9916
T	29 w=2.2900	(:)
Mx 	225.5904
Set font 	Libertinus Math;18;400;Regular;normal;;;LTR
T	2070 w=10.7640	(nil)
Mx 	236.3544
T	2758 w=8.4780	(nil)
Mx 	249.8859
T	2078 w=11.4660	(nil)
Mx 	266.4053
T	1880 w=14.8320	(nil)
Mx 	281.2373
T	13 w=3.9600	(nil)
Mx 	288.1973
Mx 	5.4000
T	3798 a=7.5780	(nil)
Mx 	295.7753
T	70 w=8.0460	(nil)
Mx 	303.8213
My 	67.2685
Set font 	Libertinus Math;14.4;400;Regular;normal;;;LTR
T	2758 w=6.7190	(nil)
Mx 	311.3144
My 	72.8485
Set font 	Libertinus Math;18;400;Regular;normal;;;LTR
T	13 w=3.9600	(nil)
Mx 	318.2744
T	3000 2990 2995 w=25.9020	(nil)
Mx 	344.1764
Mx 	5.4000
T	3798 a=7.5780	(nil)
Mx 	351.7544
T	2758 w=8.4780	(nil)
Mx 	360.2324
Mx 	5.3820
T	3799 a=7.5780	(nil)
Mx 	367.8104
Mx 	5.3820
T	3799 a=7.5780	(nil)
Mx 	380.4419
T	2078 w=11.4660	(nil)
Mx 	396.9614
Mx 	5.0760
T	9 a=6.4080	(nil)
Mx 	403.3694
T	1887 w=11.1240	(nil)
Mx 	418.5148
T	2601 w=9.4860	(nil)
Mx 	432.0222
T	1884 w=12.6360	(nil)
Mx 	444.6582
Mx 	5.0760
T	10 a=6.4080	(nil)
Mx 	451.0662
My 	63.8485
Set font 	Libertinus Math;14.4;400;Regular;normal;;;LTR
T	19 w=6.6960	(nil)
Mx 	49.7638
My 	85.1788
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	48 68 87 75 w=21.6113	(Math)
Mx 	74.0597
T	89 68 85 76 68 81 87 86 w=33.6035	(variants)
Mx 	107.6632
T	15 w=2.2900	(,)
Mx 	112.6379
T	71 76 86 83 79 68 92 w=29.3018	(display)
Mx 	141.9397
T	15 w=2.2900	(,)
Mx 	146.9143
T	48 68 87 75 48 47 w=34.4336	(MathML)
Mx 	181.3479
T	29 w=2.2900	(:)
Mx 	233.0020
My 	107.5823
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	2070 w=5.9800	(nil)
Mx 	238.9820
T	2758 w=4.7100	(nil)
Mx 	246.4697
T	2078 w=6.3700	(nil)
Mx 	255.6175
T	1880 w=8.2400	(nil)
Mx 	263.8575
T	13 w=2.2000	(nil)
Mx 	267.7242
Mx 	3.0000
T	3798 a=4.2100	(nil)
Mx 	271.9342
T	70 w=4.4700	(nil)
Mx 	276.4042
My 	103.8323
Set font 	Libertinus Math;8;400;Regular;normal;;;LTR
T	2758 w=3.7328	(nil)
Mx 	280.5670
My 	107.5823
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	13 w=2.2000	(nil)
Mx 	284.4336
T	3000 2990 2995 w=14.3900	(nil)
Mx 	298.8236
Mx 	3.0000
T	3798 a=4.2100	(nil)
Mx 	303.0336
T	2758 w=4.7100	(nil)
Mx 	307.7436
Mx 	2.9900
T	3799 a=4.2100	(nil)
Mx 	311.9536
Mx 	2.9900
T	3799 a=4.2100	(nil)
Mx 	318.9414
T	2078 w=6.3700	(nil)
Mx 	328.0892
Mx 	2.8200
T	9 a=3.5600	(nil)
Mx 	331.6492
T	1887 w=6.1800	(nil)
Mx 	340.0514
T	2601 w=5.2700	(nil)
Mx 	347.5436
T	1884 w=7.0200	(nil)
Mx 	354.5636
Mx 	2.8200
T	10 a=3.5600	(nil)
Mx 	358.1236
My 	102.5823
Set font 	Libertinus Math;8;400;Regular;normal;;;LTR
T	19 w=3.7200	(nil)
Mx 	49.7638
My 	128.6643
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	48 68 87 75 w=21.6113	(Math)
Mx 	74.0543
T	89 68 85 76 68 81 87 86 w=33.6035	(variants)
Mx 	107.6578
T	15 w=2.2900	(,)
Mx 	112.6270
T	71 76 86 83 79 68 92 w=29.3018	(display)
Mx 	144.6079
T	11 w=3.1689	(()
Mx 	147.7769
T	79 68 85 74 72 w=20.7568	(large)
Mx 	171.2129
T	73 82 81 87 w=17.0996	(font)
Mx 	188.3125
T	12 w=3.1689	())
Mx 	191.4814
T	15 w=2.2900	(,)
Mx 	196.4507
T	48 68 87 75 48 47 w=34.4336	(MathML)
Mx 	230.8842
T	29 w=2.2900	(:)
Mx 	181.2933
My 	158.9718
Set font 	Libertinus Math;18;400;Regular;normal;;;LTR
T	2070 w=10.7640	(nil)
Mx 	192.0573
T	2758 w=8.4780	(nil)
Mx 	205.5353
T	2078 w=11.4660	(nil)
Mx 	222.0013
T	1880 w=14.8320	(nil)
Mx 	236.8333
T	13 w=3.9600	(nil)
Mx 	243.7933
Mx 	5.4000
T	3798 a=7.5780	(nil)
Mx 	251.3713
T	70 w=8.0460	(nil)
Mx 	259.4173
My 	152.2218
Set font 	Libertinus Math;14.4;400;Regular;normal;;;LTR
T	2758 w=6.7190	(nil)
Mx 	266.9103
My 	158.9718
Set font 	Libertinus Math;18;400;Regular;normal;;;LTR
T	13 w=3.9600	(nil)
Mx 	273.8703
T	3000 2990 2995 w=25.9020	(nil)
Mx 	299.7723
Mx 	5.4000
T	3798 a=7.5780	(nil)
Mx 	307.3503
T	2758 w=8.4780	(nil)
Mx 	315.8283
Mx 	5.3820
T	3799 a=7.5780	(nil)
Mx 	323.4063
Mx 	5.3820
T	3799 a=7.5780	(nil)
Mx 	335.9843
T	2078 w=11.4660	(nil)
Mx 	352.4503
Mx 	5.0760
T	9 a=6.4080	(nil)
Mx 	358.8583
T	1887 w=11.1240	(nil)
Mx 	373.9823
T	2601 w=9.4860	(nil)
Mx 	387.4683
T	1884 w=12.6360	(nil)
Mx 	400.1043
Mx 	5.0760
T	10 a=6.4080	(nil)
Mx 	406.5123
My 	149.9718
Set font 	Libertinus Math;14.4;400;Regular;normal;;;LTR
T	19 w=6.6960	(nil)
Mx 	49.7638
My 	182.6958
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	48 68 87 75 w=21.6113	(Math)
Mx 	74.0491
T	89 68 85 76 68 81 87 86 w=33.6035	(variants)
Mx 	107.6526
T	15 w=2.2900	(,)
Mx 	112.6167
T	87 72 91 87 w=16.5332	(text)
Mx 	129.1499
T	15 w=2.2900	(,)
Mx 	134.1139
T	55 72 59 w=16.2793	(TeX)
Mx 	150.3932
T	29 w=2.2900	(:)
Mx 	155.3573
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	2070 w=5.9800	(nil)
Mx 	161.3373
T	2758 w=4.7100	(nil)
Mx 	168.9044
T	2078 w=6.3700	(nil)
Mx 	178.1314
T	1880 w=8.2400	(nil)
Mx 	186.3714
T	13 w=2.2000	(nil)
Mx 	190.2381
T	9 w=3.5600	(nil)
Mx 	193.7981
T	70 w=4.4700	(nil)
Mx 	198.2681
My 	179.5958
Set font 	Libertinus Math;8;400;Regular;normal;;;LTR
T	2758 w=3.7328	(nil)
Mx 	202.4309
My 	182.6958
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	13 w=2.2000	(nil)
Mx 	206.2976
T	84 74 79 w=12.0300	(nil)
Mx 	218.3276
T	9 w=3.5600	(nil)
Mx 	221.8876
T	2758 w=4.7100	(nil)
Mx 	226.5976
T	10 w=3.5600	(nil)
Mx 	230.1576
T	10 w=3.5600	(nil)
Mx 	236.5746
T	2078 w=6.3700	(nil)
Mx 	245.8017
Mx 	2.8200
T	9 a=3.5600	(nil)
Mx 	249.3617
T	1887 w=6.1800	(nil)
Mx 	257.7956
T	2601 w=5.2700	(nil)
Mx 	265.3196
T	1884 w=7.0200	(nil)
Mx 	272.3396
Mx 	2.8200
T	10 a=3.5600	(nil)
Mx 	275.8996
My 	177.6958
Set font 	Libertinus Math;8;400;Regular;normal;;;LTR
T	19 w=3.7200	(nil)
Mx 	49.7638
My 	194.6958
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	48 68 87 75 w=21.6113	(Math)
Mx 	74.0617
T	89 68 85 76 68 81 87 86 w=33.6035	(variants)
Mx 	107.6652
T	15 w=2.2900	(,)
Mx 	112.6419
T	71 76 86 83 79 68 92 w=29.3018	(display)
Mx 	141.9437
T	15 w=2.2900	(,)
Mx 	146.9203
T	55 72 59 w=16.2793	(TeX)
Mx 	163.1996
T	29 w=2.2900	(:)
Mx 	235.4820
My 	217.0992
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	2070 w=5.9800	(nil)
Mx 	241.4620
T	2758 w=4.7100	(nil)
Mx 	248.9497
T	2078 w=6.3700	(nil)
Mx 	258.0975
T	1880 w=8.2400	(nil)
Mx 	266.3375
T	13 w=2.2000	(nil)
Mx 	270.2042
T	9 w=3.5600	(nil)
Mx 	273.7642
T	70 w=4.4700	(nil)
Mx 	278.2342
My 	213.3492
Set font 	Libertinus Math;8;400;Regular;normal;;;LTR
T	2758 w=3.7328	(nil)
Mx 	282.3970
My 	217.0992
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	13 w=2.2000	(nil)
Mx 	286.2636
T	84 74 79 w=12.0300	(nil)
Mx 	298.2936
T	9 w=3.5600	(nil)
Mx 	301.8536
T	2758 w=4.7100	(nil)
Mx 	306.5636
T	10 w=3.5600	(nil)
Mx 	310.1236
T	10 w=3.5600	(nil)
Mx 	316.4614
T	2078 w=6.3700	(nil)
Mx 	325.6092
Mx 	2.8200
T	9 a=3.5600	(nil)
Mx 	329.1692
T	1887 w=6.1800	(nil)
Mx 	337.5714
T	2601 w=5.2700	(nil)
Mx 	345.0636
T	1884 w=7.0200	(nil)
Mx 	352.0836
Mx 	2.8200
T	10 a=3.5600	(nil)
Mx 	355.6436
My 	212.0992
Set font 	Libertinus Math;8;400;Regular;normal;;;LTR
T	19 w=3.7200	(nil)
Mx 	295.2916
My 	780.6177
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	20 w=4.6924	(1)
End page
Finish
